eng/submit-source-build-release-pr.sh

```diff
@@ -0,0 +1,157 @@
+   echo "--globalJson, -g      (Optional) path to the global.json file to update"
+   echo "--versionsProps, -p   (Optional) path to the Versions.props file to update"
+   echo "--setupGitAuth, -G    (Optional) set up git authentication using the gh CLI"
+   echo "--help, -h            (Optional) print this help message and exit"
+   echo
+}
+
+SHORT=t:f:v:T:B:b:g:p:Gh
+LONG=targetRepo:,forkRepo:,sdkVersion:,title:,body:,targetBranch:,globalJson:,versionsProps:,setupGitAuth,help
+
+OPTS=$(getopt --options $SHORT --long $LONG --name "$0" -- "$@")
+if [ $? != 0 ] ; then echo "Failed to parse options." >&2 ; exit 1 ; fi
+eval set -- "$OPTS"
+
+global_json_path='src/SourceBuild/tarball/content/global.json'
+versions_props_path='eng/Versions.props'
+custom_target_branch=''
+setup_git_auth=''
+
+while true ; do
+  case "$1" in
+    -h | --help )
+      print-help
+      exit 0
+      ;;
+    -G | --setupGitAuth )
+      setup_git_auth='true'
+      shift
+      ;;
+    -t | --targetRepo )
+      target_repo="$2"
+      shift 2
+      ;;
+    -f | --forkRepo )
+      fork_repo="$2"
+      shift 2
+      ;;
+    -v | --sdkVersion )
+      sdk_version="$2"
+      shift 2
+      ;;
+    -T | --title )
+      title="$2"
+      shift 2
+      ;;
+    -B | --body )
+      body="$2"
+      shift 2
+      ;;
+    -g | --globalJson )
+      global_json_path="$2"
+      shift 2
+      ;;
+    -p | --versionsProps )
+      versions_props_path="$2"
+      shift 2
+      ;;
+    -b | --targetBranch )
+      custom_target_branch="$2"
+      shift 2
+      ;;
+    -- )
+      shift
+      break
+      ;;
+    *)
+      echo "Internal error! Are you missing required arguments?"
+      exit 1
+      ;;
+  esac
+done
+
+echo "target_repo = $target_repo"
+echo "fork_repo = $fork_repo"
+echo "sdk_version = $sdk_version"
+echo "title = $title"
+echo "body = $body"
+echo "global_json_path = $global_json_path"
+echo "versions_props_path = $versions_props_path"
+echo "custom_target_branch = $custom_target_branch"
+echo "setup_git_auth = $setup_git_auth"
+
+if [[ ${setup_git_auth} == true ]]; then
+  echo "Setting up git auth"
+  gh auth setup-git
+fi
+
+if [ -z ${custom_target_branch} ]; then
+  pr_target_branch=$(echo "release/${sdk_version}" | sed 's/..$/xx/')
+else
+  pr_target_branch="${custom_target_branch}"
+fi
+
+fork_url="https://github.com/${fork_repo}"
+upstream_url="https://github.com/${target_repo}"
+
+# clone forked repo
+repo_dir="repo-${time}" # "dotnet-installer-${time}"
+git clone "${fork_url}" "${repo_dir}"
+cd "${repo_dir}"
+
+# set up upstream remote
+git remote add upstream "${upstream_url}"
+git fetch upstream
+
+# check out upstream target branch to make a new branch from
+git checkout upstream/$pr_target_branch
+
+# checkout a new branch on fork
+date=$(date +"%b%Y" | sed 's/.*/\L&/')
+new_branch_name="${sdk_version}-${date}-source-build-${time}"
+git checkout -b "${new_branch_name}"
```

Comment:
Instead of invoking git checkout twice you could combine them.

`git checkout -b ${new_branch_name} upstream/$pr_target_branch`